lib/protocol/protocol.go

```diff
@@ -121,19 +121,34 @@ var (
 	errFileHasNoBlocks    = errors.New("file with empty block list")
 )
 
+type contextLessModel interface {
+	// An index was received from the peer device
```

Comment:
Instead of repeating the comment, I'd add an interface level comment explaining that this is 1-to-1 `Model` without the initial `Connection` param.